archive_pipeline/archive.py

```diff
@@ -0,0 +1,147 @@
+"""Script for removing data older than 24 hours and saving older data to a .csv file."""
+
+import sys
+import os
+from datetime import datetime, timedelta, date
+from pytz import timezone
+
+
+from dotenv import dotenv_values
+from psycopg2 import connect
+import pandas as pd
+from boto3 import client
+
+CSV_COLUMNS = ["entry_id", "species", "temperature", "soil_moisture",
+               "humidity", "last_watered", "recording_taken", "sunlight",
+               "botanist_name", "cycle"]
+
+
+def get_database_connection(config: dict):  # pragma: no cover
+    """Establishes a connection with the PostgreSQL database."""
+    try:
+        return connect(
+            user=config['DATABASE_USERNAME'],
+            password=config['DATABASE_PASSWORD'],
+            host=config['DATABASE_IP'],
+            port=config['DATABASE_PORT'],
+            database=config['DATABASE_NAME'])
+    except ValueError as err:
+        print("Error connecting to database: ", err)
+        sys.exit()
+
+
+def get_previous_day_timestamp() -> str:
+    """Returns the timestamp of the previous day."""
+    current_datetime = datetime.now(tz=timezone("Europe/London"))
+    previous_day = current_datetime - timedelta(days=1)
+    previous_day_str = datetime.strftime(previous_day, "%Y-%m-%d %H:%M:%S")
+    return previous_day_str
+
+
+def delete_old_rows(conn, delete_timestamp: str) -> list[tuple]:  # pragma: no cover
+    """Removes rows from the Plant table that are older than a day.
+
+    Returns the deleted rows.
+    """
+    with conn.cursor() as cur:
+        print("Deleting rows from RDS.")
+        cur.execute(
+            """DELETE FROM plant
+            WHERE recording_taken < %s
+            RETURNING *;""", (delete_timestamp,))
+        deleted_rows = cur.fetchall()
+        conn.commit()
+        cur.close()
+        print("Rows deleted from RDS.")
+        return deleted_rows
+
+
+def get_deleted_rows(conn, delete_timestamp: str) -> list[tuple]:  # pragma: no cover
```

Comment:
I think the variable name is a bit confusing as the prev function suggested they've been deleted and this one is saying we're getting the already deleted rows -- maybe _get_rows_to_be_deleted_